### PR TITLE
Add build documentation to README.dev.md

### PR DESCRIPTION
**Pull request by abelsiqueira**
**Description**

Add developer documentation to be able to build the library.

**Related issues**:

- #33 

**Instructions to review the pull request**

- Check that CHANGELOG.md has been updated if necessary

Review online.

### CODE REVIEW

File: README.dev.md [hunk 1]
```diff
@@ -66,7 +126,7 @@ After a pull request is created, a Codacy test should appear. Follow the link th
 
 #### pre-commit hooks
```
Comment (fdiblen) [inline README.dev.md:127]:
```suggestion
#### pre-commit hooks (optional)
```

File: README.dev.md [hunk 2]
```diff
@@ -18,6 +18,66 @@ cd cudawrappers
...
+```shell
+sudo apt install build-essential
+```
+
+Next, you need CMake 3.17 or above. Check if you have the correct version installed with `cmake --version`.
+If your CMake version is not adequate, you can install CMake manually by downloading the latest **stable** version from the [CMake downloads page](https://cmake.org/download/).
+
+For instance, the commands below download install CMake 3.22.
+
+```shell
```
Comment (fdiblen) [inline README.dev.md:49]: can we just point to official documentation at https://cmake.org/install/ ?

File: README.dev.md [hunk 3]
```diff
@@ -18,6 +18,66 @@ cd cudawrappers
...
+cd $HOME # Or move to your preferrer location
+wget https://github.com/Kitware/CMake/releases/download/v3.22.2/cmake-3.22.2.tar.gz
+tar -zxf cmake-3.22.2.tar.gz
+./bootstrap
+make
+sudo make install
+```
+
+If you can't run `sudo` to install `cmake` - for instance if you are in a cluster without privileges - you should add the path to the CMake `bin` folder to your path.
```
Comment (fdiblen) [inline README.dev.md:58]: Normal user needs to use `CMAKE_INSTALL_PREFIX` flag. Can we point to the official documentation at https://cmake.org/cmake/help/v3.22/variable/CMAKE_INSTALL_PREFIX.html?

**Review by fdiblen — CHANGES_REQUESTED** [review]
Lookg good @abelsiqueira . I made some suggestions.

**Comment by abelsiqueira** [comment]
I have updated again

**Review by fdiblen — APPROVED** [review]
Looks good @abelsiqueira Thanks a lot! :medal_military:

**Comment by abelsiqueira** [comment]
Thanks for reviewing :stadium:

**Pull request merged**